tests/components/mqtt/test_light_template.py

```diff
@@ -320,9 +326,284 @@
     assert state.attributes.get("white_value") == 50
     assert state.attributes.get(ATTR_ASSUMED_STATE)
 
+    await common.async_turn_off(hass, "light.test")
+    mqtt_mock.async_publish.assert_called_once_with(
+        "test_light_rgb/set", "off", 2, False
+    )
+    mqtt_mock.async_publish.reset_mock()
+    state = hass.states.get("light.test")
+    assert state.state == STATE_OFF
+
+    await common.async_turn_on(hass, "light.test")
+    mqtt_mock.async_publish.assert_called_once_with(
+        "test_light_rgb/set", "on,,,,--", 2, False
+    )
+    mqtt_mock.async_publish.reset_mock()
+    state = hass.states.get("light.test")
+    assert state.state == STATE_ON
+
+    # Set color_temp
+    await common.async_turn_on(hass, "light.test", color_temp=70)
+    mqtt_mock.async_publish.assert_called_once_with(
+        "test_light_rgb/set", "on,,70,,--", 2, False
+    )
+    mqtt_mock.async_publish.reset_mock()
+    state = hass.states.get("light.test")
+    assert state.state == STATE_ON
+    assert state.attributes.get("color_temp") == 70
+
+    # Set full brightness
+    await common.async_turn_on(hass, "light.test", brightness=255)
+    mqtt_mock.async_publish.assert_called_once_with(
+        "test_light_rgb/set", "on,255,,,--", 2, False
+    )
+    mqtt_mock.async_publish.reset_mock()
+    state = hass.states.get("light.test")
+    assert state.state == STATE_ON
+    assert state.attributes.get("brightness") == 255
+
+    # Full brightness - no scaling of RGB values sent over MQTT
+    await common.async_turn_on(
+        hass, "light.test", rgb_color=[255, 128, 0], white_value=80
+    )
+    mqtt_mock.async_publish.assert_called_once_with(
+        "test_light_rgb/set", "on,,,80,255-128-0", 2, False
+    )
+    mqtt_mock.async_publish.reset_mock()
+    state = hass.states.get("light.test")
+    assert state.state == STATE_ON
+    assert state.attributes.get("white_value") == 80
+    assert state.attributes.get("rgb_color") == (255, 128, 0)
+
+    # Full brightness - normalization of RGB values sent over MQTT
+    await common.async_turn_on(hass, "light.test", rgb_color=[128, 64, 0])
+    mqtt_mock.async_publish.assert_called_once_with(
+        "test_light_rgb/set", "on,,,,255-127-0", 2, False
+    )
+    mqtt_mock.async_publish.reset_mock()
+    state = hass.states.get("light.test")
+    assert state.state == STATE_ON
+    assert state.attributes.get("rgb_color") == (255, 127, 0)
+
+    # Set half brightness
+    await common.async_turn_on(hass, "light.test", brightness=128)
+    mqtt_mock.async_publish.assert_called_once_with(
+        "test_light_rgb/set", "on,128,,,--", 2, False
+    )
+    mqtt_mock.async_publish.reset_mock()
+    state = hass.states.get("light.test")
+    assert state.state == STATE_ON
+    assert state.attributes.get("brightness") == 128
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5199018)